src/main/kotlin/com/makarytskyi/rentcar/bpp/InvocationTrackerBeanPostProcessor.kt

```diff
@@ -0,0 +1,79 @@
+package com.makarytskyi.rentcar.bpp
+
+import com.makarytskyi.rentcar.annotation.InvocationTracker
+import java.lang.reflect.InvocationHandler
+import java.lang.reflect.InvocationTargetException
+import java.lang.reflect.Method
+import java.lang.reflect.Proxy
+import kotlin.reflect.KClass
+import kotlin.reflect.full.findAnnotation
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.config.BeanPostProcessor
+import org.springframework.stereotype.Component
+
+@Component
+internal class InvocationTrackerBeanPostProcessor : BeanPostProcessor {
+
+    private val classes: MutableMap<String, KClass<*>> = HashMap()
+
+    override fun postProcessBeforeInitialization(bean: Any, beanName: String): Any {
+        val beanClass = bean::class
+        if (beanClass.findAnnotation<InvocationTracker>() != null) {
+            classes[beanName] = bean::class
+        }
+        return bean
+    }
+
+    override fun postProcessAfterInitialization(bean: Any, beanName: String): Any? {
+        val originalClassName = classes[beanName]?.simpleName
```

Comment:
Naming is kinda confusing
```suggestion
        val simpleClassName = classes[beanName]?.simpleName
```